### DISCUSSION TITLE
Regarding "[org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-4) Uncaught server error: java.lang.RuntimeException: Unable to find factory for AuthenticatorFactory: script-check_otp.js did you forget to declare it in a META-INF/services file?"

### DISCUSSION THREAD

**Discussion by dsetty25** [keycloak/keycloak#25582] [Q&A]
We are doing docker upgrade 
Installed Keycloak 22.x 
Old DB dump → new DB import
Login to CKEY console is fine, all users and clients exist in new environment
Copied the jar in providers directory

Get error below:
[org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-4) Uncaught server error: java.lang.RuntimeException: Unable to find factory for AuthenticatorFactory: script-check_otp.js did you forget to declare it in a META-INF/services file?

The json file is also proper as documented.
{
    "authenticators": [
        {
            "name": "OTP Check",
            "fileName": "check_otp.js",
            "description": "Grant access via OTP"
        }
    ]
}

The js file:

AuthenticationFlowError = Java.type("org.keycloak.authentication.AuthenticationFlowError");

Response = Java.type("javax.ws.rs.core.Response");
ResponseStatus = Java.type("javax.ws.rs.core.Response.Status");

function authenticate(context) {
	var username = user ? user.username : "anonymous";
	LOG.info(script.name + " trace auth for: " + username);
	if(null === context.getHttpRequest().getDecodedFormParameters().getFirst("otp")) {
		context.failure(AuthenticationFlowError.INVALID_CREDENTIALS, Response.status(ResponseStatus.PRECONDITION_FAILED).entity("{\"error\": \"invalid_grant\",\"error_description\": \"OTP missing\"}").type("application/json").build());
		return;
	}   
	context.success();
}

Could someone tell what needs to be done

**Comment by danielFesenmeyer** [comment]
Your JS file contains references to "javax.ws" packages, which are no longer on the classpath (mentioned [here](https://www.keycloak.org/docs/latest/release_notes/index.html#transition-from-java-ee-to-jakarta-ee)). You should replace those with "jakarta.ws".
